substrate/runtime/staking/src/lib.rs

```diff
@@ -822,14 +978,14 @@ impl<T: Trait> Module<T> {
 
 	/// Increase TotalStake by Value.
 	pub fn increase_total_stake_by(value: T::Balance) {
-		if <TotalStake<T>>::exists() {
-			<TotalStake<T>>::put(<Module<T>>::total_stake() + value);
+		if let Some(v) = <Module<T>>::total_stake().checked_add(&value) {
+			<TotalStake<T>>::put(v);
 		}
 	}
 	/// Decrease TotalStake by Value.
 	pub fn decrease_total_stake_by(value: T::Balance) {
-		if <TotalStake<T>>::exists() {
-			<TotalStake<T>>::put(<Module<T>>::total_stake() - value);
+		if let Some(v) = <Module<T>>::total_stake().checked_sub(&value) {
+			<TotalStake<T>>::put(v);
```

Comment:
Why is this conditionally applied? This seems to be a way for introducing inconsistency to this counter

Reply:
if `TotalStake` is implemented correctly, they're equivalent. however if it's not and it undercounts, leading to eventual underflow, then a panic is worse than ignoring, since an invalid `TotalStake` can be fixed in a later block, but if it gets so low that all transactions cause a panic, rescuing the chain becomes an issue of rolling back which is far harder and possibly even impractical to orchestrate.